ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecryptPioneerPayloads.java

```diff
@@ -81,11 +91,36 @@ private DecryptPioneerPayloads(ValueProvider<String> metadataLocation,
         keyStore = KeyStore.of(metadataLocation.get(), kmsEnabled.get());
       }
 
+      if (validator == null) {
+        validator = new JsonValidator();
+        byte[] data = Resources
+            .toByteArray(Resources.getResource("telemetry.pioneer-study.4.schema.json"));
+        envelopeSchema = JSONSchemaStore.readSchema(data);
+      }
+
       ObjectNode json = Json.readObjectNode(message.getPayload());
-      PrivateKey key = keyStore.getKey(message.getAttribute(Attribute.DOCUMENT_NAMESPACE));
+      validator.validate(envelopeSchema, json);
+      JsonNode payload = json.get("payload");
+
+      String encryptionKeyId = payload.get("encryptionKeyId").asText();
```

Comment:
We should probably make string constants in this class for all these literal string field names.